### ISSUE TITLE
ABI in solidity 0.6.0 does not have constant property which is causing UI libraries to think it's a non-constant method.

### ISSUE THREAD

**Issue by zemse** [ethereum/solidity#8065]
## Description
A View Method:
```
function getValue() view public returns (string memory) {
    return _value;
}
```

ABI of a view method in Solidity 0.6.0
```
{ inputs: [],
    name: 'getValue',
    outputs: [ [Object] ],
    stateMutability: 'view',
    type: 'function' },
```
While same function code in Solidity 0.5.15 generates the following ABI.
```
{ constant: true,
    inputs: [],
    name: 'getValue',
    outputs: [ [Object] ],
    payable: false,
    stateMutability: 'view',
    type: 'function' }
```

This difference in ABI is causing a transaction to send instead of a function call.
```
const currentValue = await simpleStorageInstance.functions.getValue();
// in 0.5.15 it returns the value.
// in 0.6.0 it returns tx object (which is returned when we do setValue)
```

## Environment

- Compiler version: 0.6.0
- Target EVM version (as per compiler settings): byzantium
- Framework/IDE (e.g. Truffle or Remix): [Smart Solidity Template](https://github.com/zemse/smart-solidity-template/generate)
- EVM execution environment / backend / blockchain client: ethers.js with ganache
- Operating system: macOS

## Steps to Reproduce
1. `git clone https://github.com/zemse/smart-solidity-template.git`
2. `cd smart-solidity-template`
3. `npm i` This will install solc 0.5.15
4. Go to file `/test/SimpleStorage.test.js` and before the line number 45, just add 
`console.log(simpleStorageJSON.abi)`.
5. `node compile.js && npm run test`. 
This will console the abi in 0.5.15.
6. `npm i solc@0.6.0`
7. Update solidity version in `contracts/SimpleStorage.sol`: `pragma solidity ^0.6.0;`
8. `node compile.js && npm run test`
This will console abi in 0.6.0.
Also the test `1) value should be set properly while deploying` fails because ethers library sends it as a transaction instead of a call.

**Comment by axic** [comment]
The `stateMutability` field was introduced with 0.4.16 (**2017-08-14**), which deprecated both the `constant` and `payable` fields. `stateMutability: 'view'` (what you have above) has the same effect as "constant" had previously.

This was announced multiple times. Apparently more than 2 years was not enough for tools to catch up :wink:

**Comment by zemse** [comment]
Okay, I get it. So I shall share this issue with the ethers.js team.

**Comment by axic** [comment]
Please do, that would be much appreciated.

**Comment by zemse** [comment]
It's fixed by the team, the next version of `ethers.js` will be released soon as `v4.0.41`
